### PR TITLE
Remove Memory_stats library

### PR DESCRIPTION
**Pull request by mrmr1993**
This PR removes the `Memory_stats` library, as part of the `LowFillRate` investigation.

This library was manually calling the garbage collector and dumping stats at regular intervals; previously we identified this as a source of slow-downs, but it appears that the code still remained in some places regardless.

Checklist:

- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them

### CODE REVIEW

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

File: src/lib/uptime_service/uptime_snark_worker.ml [hunk 1]
```diff
@@ -17,8 +17,7 @@ module Worker_state = struct
 
   type t = (module S)
 
-  let create ~logger : t Deferred.t =
-    Memory_stats.log_memory_stats logger ~process:"uptime service SNARK worker" ;
+  let create ~logger:_ : t Deferred.t =
```
Comment (nholland94) [inline src/lib/uptime_service/uptime_snark_worker.ml:20]: Should we just remove ~logger as an argument now?

**Review by nholland94 — APPROVED** [review]

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Pull request merged**